client/my-sites/post-type-filter/index.jsx

```diff
@@ -75,70 +72,103 @@ const PostTypeFilter = React.createClass( {
 			}
 
 			return memo.concat( {
-				count: jetpack ? null : count,
+				// Hide count in all sites mode; and in Jetpack mode for non-posts
 				key: `filter-${ status }`,
-				path: [
-					'/types',
-					query.type,
+				count: ( ! siteId || ( jetpack && query.type !== 'post' ) ) ? null : count,
+				path: compact( [
+					query.type === 'post'
+						? '/posts'
+						: '/types/' + query.type,
+					query.type === 'post' && query.author && 'my',
 					pathStatus,
 					siteSlug
-				].filter( Boolean ).join( '/' ),
+				] ).join( '/' ),
 				selected: mapPostStatus( pathStatus ) === query.status,
 				children: label
 			} );
 		}, [] );
 	},
 
 	render() {
-		const { siteId, query, jetpack } = this.props;
+		const { authorToggleHidden, jetpack, query, siteId, statusSlug } = this.props;
 		const navItems = this.getNavItems();
 		const selectedItem = find( navItems, 'selected' ) || {};
 
+		const scopes = {
+			me: this.translate( 'Me', { context: 'Filter label for posts list' } ),
+			everyone: this.translate( 'Everyone', { context: 'Filter label for posts list' } )
+		};
+
 		return (
 			<div>
-				{ query && siteId && false === jetpack && (
+				{ siteId && false === jetpack && (
 					<QueryPostCounts
 						siteId={ siteId }
 						type={ query.type } />
 				) }
 				<SectionNav
-					selectedText={ selectedItem.children }
+					selectedText={
+						<span>
+							<span>{ selectedItem.children }</span>
+							{ ! authorToggleHidden && <small>{ query.author ? scopes.me : scopes.everyone }</small> }
+						</span>
+					}
 					selectedCount={ selectedItem.count }>
-					{ query && [
```

Comment:
Do you know why this was here?

Reply:
I think it was a half-assed way of hiding components that wouldn't work if no `query` was passed. Thinking about it, I came to the conclusion that the component can't work with a `null` `query` at all, since we need at least a `type` to look up, so I opted to make it `isRequired`, and change the consumer to hide the entire component if `query` is falsey, see https://github.com/Automattic/wp-calypso/pull/18113/commits/78d13436b6df30cb7108f0a573985edf8c70f3d3#diff-427802006d2deabbea5c234b62ff4738L32